script.js

```diff
@@ -0,0 +1,124 @@
+let hours = 0;
+let minutes = 0;
+let seconds = 0;
+let milliseconds = 0;
+let timerInterval;
+let isStopwatchActive = false;
+let previousLapTime = 0;
+let lapCounter = 0;
+
+// Start the stopwatch
+function startStopwatch() {
+  if (!isStopwatchActive) {
+    isStopwatchActive = true;
+    timerInterval = setInterval(updateDisplay, 10);
+
+    // Show Stop, Lap, and Reset buttons
+    document.getElementById('start').style.display = 'none';
+    document.getElementById('stop').style.display = 'inline-block';
+    document.getElementById('lap').style.display = 'inline-block';
+    document.getElementById('lap').disabled = false; // Enable Lap button
+    document.getElementById('reset').style.display = 'inline-block';
+  }
+}
+
+// Stop the stopwatch
+function stopStopwatch() {
+  isStopwatchActive = false;
+  clearInterval(timerInterval);
+
+  // Show Start button again, hide Stop button
+  document.getElementById('start').style.display = 'inline-block';
+  document.getElementById('stop').style.display = 'none';
+
+  // Disable Lap button
+  document.getElementById('lap').disabled = true;
+}
+
+// Reset the stopwatch
+function resetStopwatch() {
+  isStopwatchActive = false;
+  clearInterval(timerInterval);
+  hours = 0;
+  minutes = 0;
+  seconds = 0;
+  milliseconds = 0;
+  previousLapTime = 0;
+  lapCounter = 0;
+  updateDisplay(true); // Reset display
+
+  // Clear laps
+  document.getElementById('laps-container').innerHTML = '';
+  document.getElementById('laps-table').style.display = 'none'; // Hide laps table
+
+  // Reset buttons: only Start button visible
+  document.getElementById('start').style.display = 'inline-block';
+  document.getElementById('stop').style.display = 'none';
+  document.getElementById('lap').style.display = 'none';
+  document.getElementById('reset').style.display = 'none';
+}
+
+// Update the stopwatch display
+function updateDisplay(isReset = false) {
+  if (!isReset) {  // Only update milliseconds if it's not a reset
```

Comment:
use time interval here and date functions to show the time like hh:mm:ss:mm not separately